src/get-options-overrides.ts

```diff
@@ -16,13 +16,28 @@ export function getOptionsOverrides({ useTsconfigDeclarationDir, cacheRoot }: IO
 		noEmitOnError: false,
 		inlineSourceMap: false,
 		outDir: normalize(`${cacheRoot}/placeholder`), // need an outdir that is different from source or tsconfig parsing trips up. https://github.com/Microsoft/TypeScript/issues/24715
-		moduleResolution: tsModule.ModuleResolutionKind.NodeJs,
 		allowNonTsExtensions: true,
 	};
 
 	if (!preParsedTsconfig)
 		return overrides;
 
+	switch (preParsedTsconfig.options.moduleResolution)
+	{
+		case tsModule.ModuleResolutionKind.Node10:
+		case tsModule.ModuleResolutionKind.Node16:
+		case tsModule.ModuleResolutionKind.NodeNext:
+			overrides.moduleResolution = preParsedTsconfig.options.moduleResolution;
+			break;
+		case tsModule.ModuleResolutionKind.Classic:
+		case undefined:
+			overrides.moduleResolution = tsModule.ModuleResolutionKind.Node10;
+			break;
+		case tsModule.ModuleResolutionKind.Bundler:
+		default:
+			overrides.moduleResolution = tsModule.ModuleResolutionKind.Node16;
+	}
+
 	if (preParsedTsconfig.options.module === undefined)
 		overrides.module = tsModule.ModuleKind.ES2015;
```

Comment:
I believe this line will never get hit now / is dead code? test coverage should confirm that

Reply:
oh I misread this, my bad. `module` vs. `moduleResolution` above